server/apps/newsletter/services.py

```
@@ -0,0 +1,71 @@
+            "reactivate_existing": True,
+            "send_welcome_email": True,
+            "utm_source": "django_website",
+            "status": "active" if is_active else "inactive"
+        }
+        
+        try:
+            response = requests.post(endpoint, headers=self.headers, json=data)
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            raise Exception(f"Failed to create Beehiiv subscriber: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)